### PR TITLE
Non homogenous flow reactor

### PR DESCRIPTION
**Pull request by gkogekar**
This is an updated version of the old PR - Non homogenous flow reactor rewrite  (https://github.com/Cantera/cantera/pull/575) created by @arghdos.  The current version is up-to-date with the latest Cantera build.

### CODE REVIEW

**Comment by bryanwweber** [comment]
@gkogekar Is there any code here that comes from the commit(s) by @arghdos? If so, those should be credited somehow. Either by cherry-picking his commits or adding him as a co-author

File: interfaces/cython/cantera/examples/surface_chemistry/1D_pfr_surfchem.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                P=reactor.thermo.P,
+                X=reactor.thermo.X,
+                t=net.time,
+                speed=reactor.speed,
+                coverages=rsurf.coverages)
+
+f, ax = plt.subplots(4, 2, figsize=(9, 16), dpi=96)
+# plot gas velocity along the flow direction
+ax[0, 0].plot(soln.t, soln.speed, color='C0')
+ax[0, 0].set_xlabel('Distance (m)')
```
Comment (rwest) [inline interfaces/cython/cantera/examples/surface_chemistry/1D_pfr_surfchem.py:58]: As a naive reader, I find it a bit confusing is the x axis representing time or distance? The variables are all called things like `t=net.time`, but the plot is labeled "Distance (m)". I get that for a PFR, one can be mapped to the other (using the speed, I guess?), but we need to help people not get confused, by choosing descriptive variable names. Are we in fact solving for residence time, or distance?

**Comment by gkogekar** [comment]
@bryanwweber This is the same code that @arghdos has written. I just picked up the source code from the old PR and rebased it with the current Cantera build. Depending on reviewer's comments, I will modify the code. So Ideally @arghdos should be added in the author's list, but I did not know how to do it here.

**Comment by ischoegl** [comment]
> @bryanwweber This is the same code that @arghdos has written. I just picked up the source code from the old PR and rebased it with the current Cantera build. Depending on reviewer's comments, I will modify the code. So Ideally @arghdos should be added in the author's list, but I did not know how to do it here.

@gkogekar ... The author's list already includes @arghdos, but he should be recognized as an author on this specific work. You can add co-authors in the commit message, e.g.
```
Implement this-and-that

Co-authored-by: Nick Curtis <replace.by@email.com>
```
Note that in general the first line of the commit message should be 72 characters at the most, followed by a blank line and a more detailed summary. Not sure what @arghdos' e-mail is at the moment.

I am also linking this comment https://github.com/Cantera/cantera/pull/575#issuecomment-470986603 as it presumably contains **relevant derivations**.

**Comment by bryanwweber** [comment]
I believe it should be the email address associated with his account, presumably the easiest is the one associated with those commits. See: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

File: interfaces/cython/cantera/examples/surface_chemistry/SiF4_NH3_mec.cti [hunk 2]
```diff
@@ -0,0 +1,564 @@
+units(length='cm', time='s', quantity='mol', act_energy='cal/mol')
```
Comment (ischoegl) [inline interfaces/cython/cantera/examples/surface_chemistry/SiF4_NH3_mec.cti:1]: CTI is no longer supported - please convert to YAML.

**Comment by skyreflectedinmirrors** [comment]
>Not sure what @arghdos' e-mail is at the moment.

arghdos AT gmail DOT com would work.

I forget exactly why I dropped this, but it's cool to see it live again :)

File: interfaces/cython/cantera/examples/surface_chemistry/1D_pfr_surfchem.py (hunk 1 above)
Comment (gkogekar) [reply interfaces/cython/cantera/examples/surface_chemistry/1D_pfr_surfchem.py:58]: The x-axis represents the distance along the reactor. The axial coordinate is considered as a time-like coordinate when using the Sundials solver. The solver typically marches in time (Here it is the axial distance instead).

**Comment by ischoegl** [comment]
@gkogekar … do you have any plans to push this over the finishing line?

**Comment by gkogekar** [comment]
> @gkogekar … do you have any plans to push this over the finishing line?

Not in the immediate future. It might take another month at the least before I get back to work on it.

**Comment by corykinney** [comment]
What is needed to finish up this pull request?

**Comment by ischoegl** [comment]
Superseded by #1490

**Pull request closed without merging**